core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -295,6 +295,10 @@ public MappedFieldType fieldType() {
         return fieldTypeRef.get();
     }
 
+    public FieldDataType fieldDataType() {
```

Comment:
I don't think this should go here. It can be a special method on ParentFieldMapper? IIRC, the users already have a ParentFieldMapper? Or is this because the warming code is generic?